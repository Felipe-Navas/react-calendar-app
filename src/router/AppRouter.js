import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import { startChecking } from '../actions/auth'
import { LoginScreen } from '../components/auth/LoginScreen'
import { CalendarScreen } from '../components/calendar/CalendarScreen'
import { PrivateRoute } from './PrivateRoute'
import { PublicRoute } from './PublicRoute'

export const AppRouter = () => {
  const dispatch = useDispatch()
  const { checking, uid } = useSelector((state) => state.auth)

  useEffect(() => {
    dispatch(startChecking())
  }, [dispatch])

  if (checking) {
    return <div>Checking...</div>
  }

  return (
    <BrowserRouter>
      <Routes>
        <Route
          exact
          path="/login"
          element={
            <PublicRoute uid={uid}>
              <LoginScreen />
            </PublicRoute>
          }
        />
        <Route
          exact
          path="/*"
          element={
            <PrivateRoute uid={uid}>
              <CalendarScreen />
            </PrivateRoute>
          }
        />
      </Routes>
    </BrowserRouter>
  )
}
