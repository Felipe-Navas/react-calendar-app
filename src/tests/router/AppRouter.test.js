import '@testing-library/jest-dom'
import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { AppRouter } from '../../router/AppRouter'

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

// store.dispatch = jest.fn()

describe('Testing the AppRouter', () => {
  test('should render correctly', () => {
    const initState = {
      auth: {
        checking: true,
      },
    }
    const store = mockStore(initState)

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('div').exists()).toBe(true)
  })

  test('should render the public route', () => {
    const initState = {
      auth: {
        checking: false,
        uid: null,
      },
    }
    const store = mockStore(initState)

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.login-container').exists()).toBe(true)
  })

  test('should render the private route', () => {
    const initState = {
      auth: {
        checking: false,
        uid: '123',
        name: 'test-name',
      },
      calendar: {
        events: [],
      },
      ui: {
        modalOpen: false,
      },
    }
    const store = mockStore(initState)

    const wrapper = mount(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    )

    expect(wrapper.find('.calendar-screen').exists()).toBe(true)
  })
})
