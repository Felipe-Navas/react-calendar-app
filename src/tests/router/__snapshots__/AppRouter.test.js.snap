// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the AppRouter should render correctly 1`] = `
<div>
  Checking...
</div>
`;

exports[`Testing the AppRouter should render the public route 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3>
        Login
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="lEmail"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="lPassword"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
    <div
      className="col-md-6 login-form-2"
    >
      <h3>
        Register
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rName"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value="Felipe"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rEmail"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value="felipe2@gmail.com"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rPassword1"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value="123456"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="rPassword2"
            onChange={[Function]}
            placeholder="Confirm password"
            type="password"
            value="123456"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Register"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
